File: ub-rider/source/c/start.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void pkg_menu();
void pkgo_menu();
void apt_menu();

void main_menu() {
    int choice;
    printf("[1] pkg\n");
    printf("[2] pkgo\n");
    printf("[3] apt\n");
    printf("Enter your choice: ");
    scanf("%d", &choice);

    switch(choice) {
        case 1:
            pkg_menu();
            break;
        case 2:
            pkgo_menu();
            break;
        case 3:
            apt_menu();
            break;
        default:
            printf("Invalid choice\n");
            break;
    }
}

void pkg_menu() {
    int pkg_choice;
    char package_name[100];

    printf("[1] install\n");
    printf("[2] search\n");
    printf("[3] update\n");
    printf("Enter your choice: ");
    scanf("%d", &pkg_choice);

    switch(pkg_choice) {
        case 1:
            printf("which package: ");
            scanf("%s", package_name);
            printf("pkg install -y %s\n", package_name);
            break;
        case 2:
            printf("which package: ");
            scanf("%s", package_name);
            printf("pkg search %s\n", package_name);
            break;
        case 3:
            printf("pkg update\n");
            break;
        default:
            printf("Invalid choice\n");
            break;
    }
}

void pkgo_menu() {
    int pkgo_choice;
    char package_name[100];

    printf("[1] -i\n");
    printf("[2] -u\n");
    printf("[3] -d\n");
    printf("Enter your choice: ");
    scanf("%d", &pkgo_choice);

    switch(pkgo_choice) {
        case 1:
            printf("which package: ");
            scanf("%s", package_name);
            printf("pkgo -i %s\n", package_name);
            break;
        case 2:
            printf("which package: ");
            scanf("%s", package_name);
            printf("pkgo -u %s\n", package_name);
            break;
        case 3:
            printf("which package: ");
            scanf("%s", package_name);
            printf("pkgo -d %s\n", package_name);
            break;
        default:
            printf("Invalid choice\n");
            break;
    }
}

void apt_menu() {
    int apt_choice;
    char package_name[100];

    printf("[1] install\n");
    printf("[2] search\n");
    printf("[3] update\n");
    printf("[4] upgrade\n");
    printf("Enter your choice: ");
    scanf("%d", &apt_choice);

    switch(apt_choice) {
        case 1:
            printf("which package: ");
            scanf("%s", package_name);
            printf("apt install -y %s\n", package_name);
            break;
        case 2:
            printf("which package: ");
            scanf("%s", package_name);
            printf("apt search %s\n", package_name);
            break;
        case 3:
            printf("apt update\n");
            break;
        case 4:
            printf("apt upgrade\n");
            break;
        default:
            printf("Invalid choice\n");
            break;
    }
}

int main() {
    main_menu();
    return 0;
}

File: setup-termux/source/python/start.py
#!/usr/bin/env python3

# ASCII Art banner
print('''
  ____            _      _  __ _     
 |  _ \  ___  ___| |_ __| |/ _(_)_ __
 | | | |/ _ \/ __| __/ _\` | |_| | '__|
 | |_| |  __/ (__| || (_| |  _| | |   
 |____/ \___|\___|\__\__,_|_| |_|_|   
''')

# Main menu function
def main_menu():
    print("[1] pkg")
    print("[2] pkgo")
    print("[3] apt")
    choice = input("Enter your choice: ")

    if choice == "1":
        pkg_menu()
    elif choice == "2":
        pkgo_menu()
    elif choice == "3":
        apt_menu()
    else:
        print("Invalid choice")

# pkg menu function
def pkg_menu():
    print("[1] install")
    print("[2] search")
    print("[3] update")
    pkg_choice = input("Enter your choice: ")

    if pkg_choice == "1":
        package_name = input("which package: ")
        print(f"pkg install -y {package_name}")
    elif pkg_choice == "2":
        package_name = input("which package: ")
        print(f"pkg search {package_name}")
    elif pkg_choice == "3":
        print("pkg update")
    else:
        print("Invalid choice")

# pkgo menu function
def pkgo_menu():
    print("[1] -i")
    print("[2] -u")
    print("[3] -d")
    pkgo_choice = input("Enter your choice: ")

    if pkgo_choice == "1":
        package_name = input("which package: ")
        print(f"pkgo -i {package_name}")
    elif pkgo_choice == "2":
        package_name = input("which package: ")
        print(f"pkgo -u {package_name}")
    elif pkgo_choice == "3":
        package_name = input("which package: ")
        print(f"pkgo -d {package_name}")
    else:
        print("Invalid choice")

# apt menu function
def apt_menu():
    print("[1] install")
    print("[2] search")
    print("[3] update")
    print("[4] upgrade")
    apt_choice = input("Enter your choice: ")

    if apt_choice == "1":
        package_name = input("which package: ")
        print(f"apt install -y {package_name}")
    elif apt_choice == "2":
        package_name = input("which package: ")
        print(f"apt search {package_name}")
    elif apt_choice == "3":
        print("apt update")
    elif apt_choice == "4":
        print("apt upgrade")
    else:
        print("Invalid choice")

# Main program
main_menu()
